### DISCUSSION TITLE
Using tmux inside a distrobox container?

### DISCUSSION THREAD

**Discussion by kastl-ars** [89luca89/distrobox#1473] [Q&A] [answered]
Hi all,

I have no deep knowledge of the internals of tmux, but it seems like when starting tmux inside a distrobox container, it instead reaches out to the "tmux server" running on the host and opens a session there. So I end up being back in the host, not inside the distrobox container. I daresay this might be due to having the host user's `/run/.../` available inside the distrobox.

Is anyone using tmux inside distrobox and has solved this without interfering with the tmux sessions running on the host?

Thanks in advance!

Kind regards
Johannes

**Comment by findurance** [comment]
I just tried this now, started tmux on both the host side and in the server side. 
Then did 'kill-server' on the host side -> killed OK.
Container side -> tmux session still on.

So, at least works for me.

Host: openSUSE Tumbleweed
Container: Ubuntu:20.04

**Reply by kastl-ars** [reply]
Sorry for the late reply. Thanks for testing.

I have a different prompt depending on when I am in a container or on the host. This makes it obvious that in my case the tmux session started inside the distrobox is connecting to the tmux server on the host.

```
if [[ -n "${CONTAINER_ID}" ]]
then
        echo "In Distrobox, not doing anything"
elif [ "$TERM" == "linux" ] || [ "$TERM" == "xterm" ] || [ "$TERM" == "xterm-256color" ]
then
[...]
``` 

If there is no tmux session running on the host, then tmux opens "inside the container", aka CONTAINER_ID is set.

If I start the tmux server inside the container (with no tmux running on the host), the next tmux session I start on the host is "running inside the container", aka CONTAINER_ID is set.

**Comment by kastl-ars** [comment, marked as answer]
I could solve this by setting a different `TMUX_TMPDIR` inside the container, so it uses a different socket for the TMUX server and thus I can have tmux in the container **and** tmux outside...
